numba/stencils/stencilparfor.py

```diff
@@ -260,13 +266,142 @@ def _mk_stencil_parfor(self, label, in_args, out_arr, stencil_ir,
+            slice_var = ir.Var(scope, mk_unique_var("$slice"), loc)
+            self.typemap[slice_var.name] = types.slice2_type
+            slice_assign = ir.Assign(slice_callexpr, slice_var, loc)
+            stmts.append(slice_assign)
+
+            # For each dimension, add setitem to set border values.
+            for dim in range(in_arr_typ.ndim):
+                # First, fill all entries with ":".
+                start_tuple_items = [slice_var] * in_arr_typ.ndim
+                last_tuple_items = [slice_var] * in_arr_typ.ndim
+
+                # Handle the border at the start of the index range.
+                # ---- Generate call to slice func.
+                sig = self.typingctx.resolve_function_type(slice_fn_ty,
+                                                           (types.intp,types.intp),
+                                                           {})
+                si = start_inds[dim]
+                assert(isinstance(si, (int, ir.Var)))
+                si_var = ir.Var(scope, mk_unique_var("$start_ind"), loc)
+                self.typemap[si_var.name] = types.intp
+                if isinstance(si, int):
+                    si_assign = ir.Assign(ir.Const(si, loc), si_var, loc)
+                else:
+                    si_assign = ir.Assign(si, si_var, loc)
+                stmts.append(si_assign)
+
+                slice_callexpr = ir.Expr.call(func=slice_func_var,
+                                              args=(zero_index_var, si_var),
+                                              kws=(),
+                                              loc=loc)
+                self.calltypes[slice_callexpr] = sig
+                # ---- Generate slice var
+                start_slice_var = ir.Var(scope, mk_unique_var("$slice"), loc)
+                self.typemap[start_slice_var.name] = types.slice2_type
+                slice_assign = ir.Assign(slice_callexpr, start_slice_var, loc)
+                stmts.append(slice_assign)
+
+                start_tuple_items[dim] = start_slice_var
+                start_ind_var = ir.Var(scope, mk_unique_var(
+                    "$start_index_tuple_var"), loc)
+                self.typemap[start_ind_var.name] = types.containers.UniTuple(
+                    types.slice2_type, ndims)
+                tuple_call = ir.Expr.build_tuple(start_tuple_items, loc)
+                tuple_assign = ir.Assign(tuple_call, start_ind_var, loc)
+                stmts.append(tuple_assign)
+
+                setitem_call = ir.SetItem(out_arr, start_ind_var, zero_var, loc)
+                self.calltypes[setitem_call] = signature(
+                                                types.none, self.typemap[out_arr.name],
+                                                self.typemap[start_ind_var.name],
+                                                self.typemap[out_arr.name].dtype
+                                                )
+                stmts.append(setitem_call)
```

Comment:
This block seems to be almost identical to the one below, apart from `start_inds` vs `last_inds` and `zero_index_var` vs `si_var`, perhaps make it into a local function and call it twice?

Reply:
Done.